error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/IsInstanceUsage.java

```diff
@@ -0,0 +1,51 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import static com.google.errorprone.BugPattern.LinkType.CUSTOM;
+import static com.google.errorprone.BugPattern.SeverityLevel.SUGGESTION;
+import static com.google.errorprone.BugPattern.StandardTags.STYLE;
+import static com.sun.source.tree.Tree.Kind.INSTANCE_OF;
+import static com.sun.source.tree.Tree.Kind.LAMBDA_EXPRESSION;
+import static tech.picnic.errorprone.bugpatterns.util.Documentation.BUG_PATTERNS_BASE_URL;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.LambdaExpressionTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.matchers.Description;
+import com.sun.source.tree.InstanceOfTree;
+import com.sun.source.tree.LambdaExpressionTree;
+
+/** A {@link BugChecker} that aligns usages of T.class::isInstance. */
+@AutoService(BugChecker.class)
+@BugPattern(
+    summary = "Use Class::isInstance where possible",
+    link = BUG_PATTERNS_BASE_URL + "IsInstanceUsage",
+    linkType = CUSTOM,
+    severity = SUGGESTION,
+    tags = STYLE)
+public final class IsInstanceUsage extends BugChecker implements LambdaExpressionTreeMatcher {
```

Comment:
As this is quite specific to lambda's, we could update the name to reflect that. Maybe something like `LambdaIsInstanceUage` or `IsInstanceLambdaUsage`?